### PR TITLE
Stac 6356 short lived processes

### PR DESCRIPTION
**Pull request by zandrewitte**

### CODE REVIEW

File: checks/process_test.go [hunk 1]
```diff
@@ -935,6 +955,97 @@ func TestRateCalculation(t *testing.T) {
...
+
+	// wait for the shortlived qualifier seconds
+	time.Sleep(cfg.ShortLivedProcessQualifierSecs)
+
+	// second run with filled in cache; expect all processes.
+	secondRun := Process.fmtProcesses(cfg, cur, containers, syst2, syst1, lastRun)
+	assert.Equal(t, 4, len(secondRun), "Processes should contain 4 elements")
+	assert.Equal(t, 4, Process.cache.ItemCount(), "Cache should contain 4 elements")
+
+	// delete pid 4 from the process map, expect it to be excluded from the cache + process list
```
Comment (TizianoPerrucci) [inline checks/process_test.go:1001]: excluded only from the process list

File: checks/process_test.go [hunk 2]
```diff
@@ -935,6 +955,97 @@ func TestRateCalculation(t *testing.T) {
...
+	delete(cur, 4)
+	thirdRun := Process.fmtProcesses(cfg, cur, containers, syst2, syst1, lastRun)
+	assert.Equal(t, 3, len(thirdRun), "Processes should contain 4 elements")
+	assert.Equal(t, 4, Process.cache.ItemCount(), "Cache should contain 4 elements")
+
+	// wait 2 seconds
+	time.Sleep(2 * time.Second)
```
Comment (TizianoPerrucci) [inline checks/process_test.go:1008]: we can use a multiplier of `ProcessCacheDuration` here, update comment

File: config/config.go [hunk 3]
```diff
@@ -202,6 +209,13 @@ func NewDefaultAgentConfig() *AgentConfig {
 		EnableIncrementalPublishing:          true,
 		IncrementalPublishingRefreshInterval: 1 * time.Minute,
 
+		// Process Cache Expiration duration
+		ProcessCacheDuration: 5 * time.Minute,
+
+		// ShortLived process filtering
+		EnableShortLivedProcessFilter:  true,
```
Comment (TizianoPerrucci) [inline config/config.go:216]: this config is never used ?

File: checks/process_test.go [hunk 4]
```diff
@@ -998,48 +1012,121 @@ func TestProcessCache(t *testing.T) {
...
 			name: fmt.Sprintf("Should not filter a process that has been observed longer than the short-lived qualifier "+
 				"duration: %d", cfg.ShortLivedProcessQualifierSecs),
-			expected:      false,
-			firstObserved: now.Add(-2 * time.Minute).Unix(),
+			prepCache: func() {
+				fp := cur[1]
+				processID := createProcessID(fp.Pid, fp.CreateTime)
+				cachedProcess := &ProcessCache{
+					Process:       fp,
+					FirstObserved: now.Add(-5 * time.Minute).Unix(),
```
Comment (TizianoPerrucci) [inline checks/process_test.go:1064]: wouldn't be better to make this relative to the `lastRun` ?

File: checks/process_test.go [hunk 5]
```diff
@@ -998,48 +1012,121 @@ func TestProcessCache(t *testing.T) {
...
 			name: fmt.Sprintf("Should filter a process that has not been observed longer than the short-lived qualifier "+
 				"duration: %d", cfg.ShortLivedProcessQualifierSecs),
-			expected:      true,
-			firstObserved: now.Add(-2 * time.Second).Unix(),
+			prepCache: func() {
+				fp := cur[1]
+				processID := createProcessID(fp.Pid, fp.CreateTime)
+				cachedProcess := &ProcessCache{
+					Process:       fp,
+					FirstObserved: now.Add(-5 * time.Second).Unix(),
```
Comment (TizianoPerrucci) [inline checks/process_test.go:1081]: wouldn't be better to make this relative to the `lastRun` ?

File: checks/process_common.go [hunk 6]
```diff
@@ -345,5 +346,13 @@ func isProcessShortLived(shortLivedProcessFilterEnabled bool, firstObserved int6
 		return false
 	}
 
+	// process is filtered due to it's short-lived nature, let's log it on trace level
+	log.Debugf("Filter process: %s based on it's short-lived nature; "+
```
Comment (TizianoPerrucci) [inline checks/process_common.go:350]: nice

File: checks/process_common.go [hunk 7]
```diff
@@ -32,7 +34,7 @@ const (
 // returns a function to filter processes in blacklist based on the configuration provided
```
Comment (TizianoPerrucci) [inline checks/process_common.go:34]: considering filtering is all happening in one place now, I would update the comment

Comment (zandrewitte) [reply checks/process_common.go:34]: Ah yes I missed that!

**Pull request merged**